### ISSUE TITLE
Fix: Add random_state to train_test_split calls for reproducible ML notebook results

### ISSUE THREAD

**Issue by Primavat** [Niketkumardheeryan/ML-CaPsule#1407]
## Description

Several machine learning notebooks in the repository use train_test_split() without specifying a random_state parameter.

Because of this, the train-test split changes every time the notebook is executed, which causes:

- inconsistent accuracy scores,
- non-reproducible outputs,
- unstable model evaluation,
- and confusion for beginners following the notebooks.

For an educational ML repository, reproducibility is extremely important so that learners can consistently obtain the same results while studying or experimenting with the notebooks.

### Current Behavior

- Example of the current implementation:

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)

Since no random_state is provided, the dataset split changes on every execution.

### Proposed Solution

Add a fixed random_state value (random_state=42) to all affected train_test_split() calls.

- Updated implementation:

X_train, X_test, y_train, y_test = train_test_split(
    X,
    y,
    test_size=0.2,
    random_state=42
)
### Notebooks Targeted

The fix will be applied to the affected notebooks where reproducibility is currently missing, including notebooks related to:

- Heart Disease Prediction
- Diabetes ML notebooks
- Flood Prediction
- Language Detection
- Startup Profit Prediction
- SVM examples
- GridSearchCV examples

and other confirmed notebooks with missing random_state.

### Why This Fix Is Important

Adding random_state improves:

- reproducibility of experiments
- consistency of notebook outputs
- debugging reliability
- educational clarity for beginners
- overall ML workflow stability

This is a standard best practice in machine learning workflows and educational repositories.

## Additional Notes

Only reproducibility-related changes will be made.
No dataset, model architecture, or training logic will be modified.
Notebook outputs will be cleared before submission to keep the PR clean.

I would like to work on this issue under GSSoC 2026 open source contribution.

**Comment by Sibam-Paul** [comment]
feel free to assign it using /assign

**Comment by Sibam-Paul** [comment]
see the pr #1409 to see the changes to make 

<img width="622" height="765" alt="Image" src="https://github.com/user-attachments/assets/414b81e5-e47e-46ab-8d44-2ae67e16cc36" />
